tests/foreman/ui/test_host.py

```diff
@@ -172,13 +173,18 @@ def setUpClass(cls):
         resource_url = u'qemu+ssh://root@{0}/system'.format(
             settings.compute_resources.libvirt_hostname
         )
-        cls.computeresource = entities.LibvirtComputeResource(
-            url=resource_url
-        ).search()
-        if len(cls.computeresource) >= 1:
-            cls.computeresource = cls.computeresource[0]
-            cls.computeresource.location = [cls.loc]
-            cls.computeresource.organization = [cls.org]
+        comp_res = entities.LibvirtComputeResource().search()
+        # Exclude everything but Libvirt compute resources from search result
+        if len(comp_res) >= 1:
+            comp_res = [
+                res for res in comp_res
+                if res.provider == 'Libvirt' and res.url == resource_url
+            ]
+        if len(comp_res) >= 1:
+            cls.computeresource = entities.LibvirtComputeResource(
+                id=comp_res[0].id).read()
+            cls.computeresource.location.append(cls.loc)
+            cls.computeresource.organization.append(cls.org_)
             cls.computeresource = cls.computeresource.update(
                 ['location',
```

Comment:
This and the next line if joined would make it more readable. Or move open [ to the line above and the closing together with the closing ).